static/js/src/modules/header-search.js

```diff
@@ -0,0 +1,11 @@
+const handleSearch = () => {
+  document.querySelector('[data-js="search-form"]').addEventListener('submit', e => {
+    e.preventDefault();
+    const text = document.querySelector('[data-js="search-form-text"]').value;
+    const re = new RegExp(' ', 'g');
+    const url = '/q/' + text.replace(re, '/');
+    window.location.pathname = url;
```

Comment:
This is because of the URL format we use for searches, we use /q/queryhere and there isn't a way to make the form redirect to those kinds of URLs without JS (this is copied over from blues browser).

Reply:
Is there a reason why that URL scheme can't change?

I'd much prefer we follow a `/search?q={term}` type URL scheme, as we do in our searches elsewhere, and especially if it means we can avoid a JS solution.

We can then add redirects for the old search URL scheme to the new one if it's important.

Reply:
If we can map the redirects from jujucharms.com to the new format I guess we can change it, but I left it as it was for compatibility.

Reply:
Since we're going to be redirecting from jujucharms.com anyway, I'd suggest it should presumably be trivial to do this redirect too, whether we're doing the redirects in Pyramid, Flask or Apache.

In which case, if it's not too much trouble, would you be happy to change the scheme and removing the JS?

Or if that's difficult, then I'm also happy to merge this as-is and then switch over the scheme in a follow-up PR instead.

Reply:
It's fine, I'll do it in this branch.

Reply:
Done.